packages/lambda-powertools-logger/index.d.ts

```diff
@@ -0,0 +1,51 @@
+import CorrelationIds from "@dazn/lambda-powertools-correlation-ids";
+export = dazn__lambda_powertools_logger;
+declare enum LogLevels {
+  DEBUG = 20,
+  INFO = 30,
+  WARN = 40,
+  ERROR = 50,
+}
+
+declare class dazn__lambda_powertools_logger {
+  correlationIds: CorrelationIds;
+  originalLevel: string;
+  level: string;
+  constructor(argument?: { correlationIds?: CorrelationIds; level?: string });
+
+  appendError<T>(params: T, err: Error): T & Error;
+
+  debug(message: string, params?: any): void;
+
+  enableDebug(): () => void;
+
+  error(message: string, err?: Error): void;
+  error(message: string, params?: any, err?: Error): void;
+
+  info(message: string, params?: any): void;
+
+  isEnabled(level: number): Boolean;
```

Comment:
Can we use `LogLevel` as a type in here instead of `number`?

Reply:
I'm not sure I can. I wrote the Loglevel interface down planning on doing like you suggested. But the orignal js just exports the class as a default. So in staying true to the js, I can't export the Loglevel enum for people to use.

Reply:
That's fine. How about something like this?

```
declare type LogLevels = {
    DEBUG: 20,
    INFO: 30,
    WARN: 40,
    ERROR: 50,
}

declare type ValueOf<T> = T[keyof T];

...

isEnabled(level: ValueOf<LogLevels>): Boolean;
```

Or even something as simple as:
```
isEnabled(level: 20 | 30 | 40 | 50): Boolean;
```

Feels like it would be helpful to be as explicit as possible.

Reply:
Smart, I like that!

Reply:
Ok all done. Sorry I was working on a few other things. I went for the 
`isEnabled(level: 20 | 30 | 40 | 50): Boolean;`
Since if people are using an IDE, it gives a better answer on what they can pick.